server/helpers/facebook.js

```diff
@@ -1,50 +1,103 @@
-const Request = require('request')
+const request = require('request')
+const server = require('../').hapi
 const log = require('./logger')
 const facebookConfig = require('../../config').facebook
 
 const facebook = {}
 
-facebook.debugToken = function (facebookUserId, facebookUserToken, cb) {
-  const url = 'https://graph.facebook.com/debug_token?input_token=' + facebookUserToken + '&access_token=' + facebookConfig.clientId + '|' + facebookConfig.clientSecret
+facebook.verifyToken = (facebookUserId, facebookUserToken) => {
+  return new Promise((resolve, reject) => {
+    const url = `https://graph.facebook.com/debug_token?input_token=${facebookUserToken}&access_token=${facebookConfig.clientId}|${facebookConfig.clientSecret}`
 
-  Request.get(url, {
-    json: true
-  },
-  (error, response, result) => {
-    if (error || response.statusCode !== 200) {
-      log.warn({err: error, facebookConfig: facebookConfig, response: response})
-      return cb(error || {err: response.statusCode, message: response.statusMessage})
-    }
+    request.get(url, { json: true }, (error, response, result) => {
+      if (error || response.statusCode !== 200) {
+        log.warn({ error, facebookConfig, response: response.statusMessage })
+        return reject('invalid facebook token')
+      }
 
-    const isValid = !(!result.data || result.data.app_id !== facebookConfig.clientId || result.data.user_id !== facebookUserId)
-
-    if (!isValid) {
-      log.warn(
-        {
-          'result-app-id': result.data.app_id,
-          'config-app-id': facebookConfig.clientId,
-          'result-user-id': result.data.user_id,
-          'requested-user-id': facebookUserId
-        },
-        'invalid facebook login!'
-      )
-    }
+      const isValid = !(!result.data || result.data.app_id !== facebookConfig.clientId || result.data.user_id !== facebookUserId)
```

Comment:
You should try not to use negative expressions, I feel like it makes it harder to read them. 
From what I understand every single one of those 3 clauses must verify in order for the toke to be valid. Thus, the `and` expression should be the one use, not `or`.

Correct  me if I'm wrong, please :)

`(result.data && result.data.app_id === facebookConfig.clientId && result.data.user_id === facebookUserId)`